src/app/g3w-ol/controls/querybydrawpolygoncontrol.js

```diff
@@ -0,0 +1,228 @@
+     * @param layer
+     */
+    onSelectlayer(layer) {
+      if (
+        layer.isSelected()
+      ) {
+        this.setSelectedLayer(layer);
+        this.setEnable(layer.isFilterable());
+      } else {
+        this.setSelectedLayer(null);
+        this.setEnable(true);
+      }
+      this.toggle(this.isToggled() && this.getEnable());
+    },
+    enabled: true,
+    layers,
+    help: {
+      title:"sdk.mapcontrols.querybybbox.help.title",
+      message:"sdk.mapcontrols.querybybbox.help.message",
+    }
+  };
+
+  this.unwatches = [];
+
+  BaseQueryPolygonControl.call(this, _options);
+
+  // feature used to store feature drawend
+  this.feature = null;
+};
+
+ol.inherits(QueryByDrawPolygonControl, BaseQueryPolygonControl);
+
+const proto = QueryByDrawPolygonControl.prototype;
+
+/**
+ * @param {ol.Map} map
+ */
+proto.setMap = function(map) {
+  
+  BaseQueryPolygonControl.prototype.setMap.call(this, map);
+
+  const eventKey = this._interaction.on('drawend', throttle(evt => {
+    this.feature = evt.feature;
+    this.dispatchEvent({ type: 'drawend', feature: this.feature });
+
+    this.runSpatialQuery();
+
+    if (this._autountoggle) {
+      this.toggle();
+    }
+  }));
+
+  this.setEventKey({
+    eventType: 'drawend',
+    eventKey
+  });
+};
+
+/**
+ * Check visibiliy of control
+ * @param layers
+ * @returns {boolean}
+ */
+proto.checkVisibile = function(layers) {
+  let visible;
+  // if no layer
+  if (!layers.length) visible = false;
+  else {
+    // get all layers filterable
+    const filterableLayers = layers.filter(layer => layer.isFilterable());
+    visible = filterableLayers.length > 0;
+  }
+  return visible;
+};
+
+
+/**
+ * @param { unknown | null } layer
+ *
+ * @since 3.8.0
+ */
+proto.listenLayersVisibilityChange = function() {
+  this.unwatches.forEach(unwatch => unwatch());
+  this.unwatches.splice(0);
+  this.layers.forEach(layer => {
+    this.unwatches.push(
+      this.watchLayer(() =>  layer.state.visible, visible => {
+        // check if a selectedLayer i set
+        if (null !== this.selectedLayer) {
+          // enable control only if current changed visible layer is true or
+          // if at least one layer (not selected) is visible
+          this.setEnable(this.isSelectedLayerVisible());
+        } else {
+          this.setEnable(this.isThereVisibleLayers());
+        }
+      }));
+  });
+};
+
+/**
+ * @since 3.8.0
+ * @param layer
+ */
+proto.handleAddExternalLayer = function(layer, unWatches) {
+  unWatches[layer.name].push(
+    this.watchLayer(
+      () => layer.selected,                                    // watch `layer.selected` property
+      selected => {
+        if (true === selected) {
+          this.setSelectedLayer(layer);
+          this.setEnable(layer.visible);
+        } else {
+          this.setSelectedLayer(null);
+          this.setEnable(this.isThereVisibleLayers());
+          this.toggle(this.isToggled() && this.getEnable());
+        }
```

Comment:
written like so could be equivalent? (I'm specifically referring to the last line `this.toggle(...)`)

```suggestion
      selected => {
        this.setSelectedLayer(true === selected ? layer : null);
        this.setEnable(true === selected ? layer.visible : this.isThereVisibleLayers());
        this.toggle(this.isToggled() && this.getEnable());
      })
```